src/main/java/org/datadog/jmxfetch/Instance.java

```diff
@@ -263,9 +263,26 @@ private void getMatchingAttributes() {
         LOGGER.info("Found " + matchingAttributes.size() + " matching attributes");
     }
 
-    private void refreshBeansList() throws IOException {
+    public LinkedList<String> getBeansScopes(){
+        if(this.beanScopes == null){
+            this.beanScopes = Configuration.getGreatestCommonScopes(configurationList);
+        }
+        return this.beanScopes;
+    }
 
-        this.beans = connection.queryMBeans();
+    private void refreshBeansList() throws IOException {
+        this.beans = new HashSet<ObjectInstance>();
+        try {
+            LinkedList<String> beanScopes = getBeansScopes();
+            for (String scope : beanScopes) {
+                ObjectName name = new ObjectName(scope);
+                this.beans.addAll(connection.queryMBeans(name));
+            }
+        }
+        catch (Exception e) {
+            e.printStackTrace();
```

Comment:
Bit of a nitpick, but could we also log an error so that the exception is not completely silent when JMXFetch's stderr is not redirected to a log?
Something like `LOGGER.error("Unable to compute common bean scope, querying all beans as a fallback")`